content.go

```diff
@@ -273,6 +274,9 @@ func Fetch(ctx context.Context, target ReadOnlyTarget, reference string, opts Fe
 		}
 		rc, err := target.Fetch(ctx, desc)
 		if err != nil {
+			if errors.Is(err, errdef.ErrNotFound) && descriptor.IsForeignLayer(desc) {
```

Comment:
By analyzing, we don't need to consider foreign layers in `oras.Fetch` as we cannot fetch a foreign layer from a reference string but a descriptor.

Let's revert the changes in the `content.go`.

Reply:
Reverted.